pkg/operator/controller/extension/add.go

```diff
@@ -18,22 +23,39 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 
+	apiextensions "github.com/gardener/gardener/pkg/api/extensions"
 	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap"
+	"github.com/gardener/gardener/pkg/controllerutils"
 	"github.com/gardener/gardener/pkg/controllerutils/mapper"
+	"github.com/gardener/gardener/pkg/extensions"
 	"github.com/gardener/gardener/pkg/operator/controller/extension/admission"
 	"github.com/gardener/gardener/pkg/operator/controller/extension/controllerregistration"
-	"github.com/gardener/gardener/pkg/operator/controller/extension/runtime"
+	extensionruntime "github.com/gardener/gardener/pkg/operator/controller/extension/runtime"
 	operatorpredicate "github.com/gardener/gardener/pkg/operator/predicate"
 	"github.com/gardener/gardener/pkg/utils/oci"
 )
 
 // ControllerName is the name of this controller.
 const ControllerName = "extension"
 
+type extensionKind struct {
+	objectKind        string
+	object            client.Object
+	newObjectListFunc func() client.ObjectList
+}
```

Comment:
`extensionKind` seems like a surprising name for what is included. `extension` seems like better suited, but feel free to decide for an even better one as long as you find something better than `extensionKind`.